lib/devicetrust/native/device_windows.go

```diff
@@ -511,6 +528,161 @@ func solveTPMEnrollChallenge(
 	}, nil
 }
 
+// activateCredentialInElevated child uses `runas` to trigger a child process
+// with elevated privileges. This is necessary because the process must have
+// elevated privileges in order to invoke the TPM 2.0 ActivateCredential
+// command.
+func activateCredentialInElevatedChild(
+	encryptedCredential attest.EncryptedCredential,
+) ([]byte, error) {
+	exe, err := os.Executable()
+	if err != nil {
+		return nil, trace.Wrap(err, "determining current executable path")
+	}
+	cwd, err := os.Getwd()
+	if err != nil {
+		return nil, trace.Wrap(err, "determining current working directory")
+	}
+
+	// Clear up the results of any previous credential activation
+	if err := os.Remove(todoCredentialActivationPath); err != nil {
+		err := trace.ConvertSystemError(err)
+		if !trace.IsNotFound(err) {
+			return nil, trace.Wrap(err, "clearing previous credential activation results")
+		}
+	}
+
+	// Assemble the parameter list. We encoded any binary data in base64.
+	// The list is then converted to a space delimited string.
+	// These parameters cause `tsh` to invoke HandleActivateCredential.
+	params := strings.Join([]string{
+		"-d", // enable debug mode so relevant logs appear in the new window
```

Comment:
This stands out a bit, I wouldn't trust debug-level logs to not start dumping loads of useless info over time.

Reply:
Yeah - I'm split on this. The window opens and closes almost immediately /unless/ an error occurs. At which point, the debug logs are going to be of value. I suppose I could pass down the value of debug mode to this code and only pass "-d" to the "child" if it's been passed to the "parent".

Reply:
I like the idea of piping `-d` from the parent, +1.